### PR TITLE
Let PartitionProcessor and Consensus module use restate_wal_protocol::Envelope

### PR DESCRIPTION
**Pull request by tillrohrmann**
This commit removes the partition Command and replaces it with the
restate_wal_protocol::Envelope. This is a preparational step for removing
the Consensus module and replacing it with Bifrost.

### CODE REVIEW

File: crates/worker/src/partition/action_effect_handler.rs [hunk 1]
```diff
@@ -57,16 +83,39 @@ impl ActionEffectHandler {
...
+        Header {
+            ack_mode: AckMode::None,
+            dest: Destination::Processor { partition_key },
+            source: Source::Processor {
+                partition_id: self.partition_id,
+                partition_key: Some(partition_key),
+                leader_epoch: self.leader_epoch,
+                // todo: Add support for deduplicating self proposals
+                sequence_number: None,
+                node_id: NodeId::my_node_id().expect("NodeId should be set").id(),
```
Comment (AhmedSoliman) [inline crates/worker/src/partition/action_effect_handler.rs:117]: ❤️

File: crates/worker/src/partition/state_machine/mod.rs [hunk 2]
```diff
@@ -63,7 +59,7 @@ impl<Codec: RawEntryCodec> StateMachine<Codec> {
         is_leader: bool,
     ) -> Result<InterpretationResult<Transaction<TransactionType>, Collector>, Error> {
         // Handle the command, returns the span_relation to use to log effects
-        let command_type = command.type_human();
+        let command_type = command.name();
```
Comment (AhmedSoliman) [inline crates/worker/src/partition/state_machine/mod.rs:62]: ❤️

File: crates/worker/src/services.rs [hunk 3]
```diff
@@ -148,3 +155,11 @@ where
         Ok(())
     }
 }
+
+fn create_header(partition_key: PartitionKey) -> Header {
+    Header {
+        source: Source::ControlPlane {},
```
Comment (AhmedSoliman) [inline crates/worker/src/services.rs:161]: Very nice!

**Review by AhmedSoliman — APPROVED** [review]
Very impressive work @tillrohrmann. 🍾 

LGTM.

File: crates/ingress-dispatcher/src/lib.rs [hunk 4]
```diff
@@ -338,12 +334,27 @@ impl IngressDispatcherOutput {
...
+            AckMode::Ack
+        };
+
+        let header = Header {
+            source: Source::Ingress {
+                node_id: from_node_id,
+                sequence_number: msg_index,
+                dedup_key: deduplication_source,
+                // todo: Retrieve proper node config version
+                nodes_config_version: Version::MIN,
```
[the remark this answers is not in the document]
Comment (tillrohrmann) [reply crates/ingress-dispatcher/src/lib.rs:349]: True, I'll remove `node_id` from `DispatcherLoopHandler`.

One option for sharing the metdata could be to make it a task local key of the `TaskCenter`.

**Comment by tillrohrmann** [comment]
Thanks a lot for the quick review @AhmedSoliman. Merging this PR now.

**Pull request merged**